<launch>
  <arg name="map_path" default="$(find-pkg-share points_map_loader)/pcd/pointcloud_map.pcd"/>
  <arg name="rviz" default="true"/>
  <arg name="min_range" default="1.0"/>
  <arg name="max_range" default="200.0"/>
  <arg name="downsample_leaf_size" default="3.0"/>
  <arg name="transformation_epsilon" default="0.01"/>
  <arg name="max_correspondence_distance" default="1.0"/>
  <arg name="euclidean_fitness_epsilon" default="0.1"/>
  <arg name="ransac_outlier_rejection_threshold" default="1.0"/>
  <arg name="max_iteration" default="100"/>
  <arg name="omp_num_thread" default="4"/>
  <arg name="map_frame_id" default="map"/>
  <arg name="base_frame_id" default="base_link"/>
  <arg name="registration_type" default="FAST_GICP"/>
  <arg name="gicp_num_thread" default="4"/>

  <include file="$(find-pkg-share points_map_loader)/launch/points_map_loader.launch.xml">
    <arg name="map_path" value="$(var map_path)"/>
    <arg name="rviz" value="false"/>
  </include>

  <!--node pkg="tf" type="static_transform_publisher" name="velodyne_transform_publisher" args="0.0 0 0.0 0 0 0 /base_link /velodyne 10" /-->

  <node pkg="icp_localization" exec="icp_localization_node" name="icp_localization_node">
    <remap from="points_map" to="points_map"/>
    <remap from="points_raw" to="velodyne_points"/>
    <param name="min_range" value="$(var min_range)"/>
    <param name="max_range" value="$(var max_range)"/>
    <param name="downsample_leaf_size" value="$(var downsample_leaf_size)"/>
    <param name="transformation_epsilon" value="$(var transformation_epsilon)"/>
    <param name="max_correspondence_distance" value="$(var max_correspondence_distance)"/>
    <param name="euclidean_fitness_epsilon" value="$(var euclidean_fitness_epsilon)"/>
    <param name="ransac_outlier_rejection_threshold" value="$(var ransac_outlier_rejection_threshold)"/>
    <param name="max_iteration" value="$(var max_iteration)"/>
    <param name="omp_num_thread" value="$(var omp_num_thread)"/>
    <param name="map_frame_id" value="$(var map_frame_id)"/>
    <param name="base_frame_id" value="$(var base_frame_id)"/>
    <param name="registration_type" value="$(var registration_type)"/>
    <param name="gicp_num_thread" value="$(var gicp_num_thread)"/>
  </node>

    <node pkg="rviz2" exec="rviz2" name="rviz2" args="-d $(find-pkg-share icp_localization)/rviz/icp.rviz"/>
</launch>
